### ISSUE TITLE
Twisted's tox file should support specifying the different reactors

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8269]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8269|
|Type|enhancement|
|Created|2016-04-03 15:27:49Z|
|Branch|https://github.com/twisted/twisted/tree/tox-reactors-8269|

Right now it's just ```py27-tests``` or whatever -- it should be something like ```py27-tests-epoll``` and such (using the "platform" key to disable them on platforms), IDEALLY -- but it looks like that due to tox limitations, it'll have to be an environment variable, or we duplicate the trial command with each of the settings, for both -tests and -coverage (which isn't great).

I think the environment variable is the best right now, to make tox usable on the buildbots.

<details><summary>Searchable metadata</summary>

```
trac-id__8269 8269
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__None None
branch__branches_tox_reactors_8269 branches/tox-reactors-8269
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1459697269556373 1459697269556373
changetime__1460085145836671 1460085145836671
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

(In [47133]) Branching to tox-reactors-8269.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl commented|
|-|-|

This adds it as a environment variable -- ```TWISTED_REACTOR``` in the environment is given to the ```--reactor``` trial flag.

Builders spun, please review.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban set owner to @hawkowl|
|-|-|

I think that if you are happy with this, we should merge it.

Are the developers supposed to run the same tox file and set the TWISTED_REACTOR env var to run some tests? If yes, is this documented?

... even if this is not documented, I hope that this is just a transition phase and in the end we will have all logic for building and running tests encapsulated in the repo.

right now, the other part of this change is in twisted-infra/braid ...and there it can be merged without review... so I would say that since this is devtools part of twisted, we should also merge it without review.

thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl set status to `closed`|
|-|-|

(In [47181]) Merge tox-reactors-8269: Add the ability to test specific reactors using the tox config

Author: hawkowl
Reviewer: adiroiban
Fixes: [#8269](https://github.com/twisted/twisted/issues/8269)
